### DISCUSSION TITLE
How to read from array of low cardinality column?

### DISCUSSION THREAD

**Discussion by kant777** [ClickHouse/ch-go#22] [Q&A]
How to read from array of low cardinality string column?

Below is the parts of the code I used to Read LowCardinality Column
```
tagsKey = proto.ColStr{}
tagsKeyLC = proto.ColLowCardinality{
	Index: &tagsKey,
}
tagsKeyArr = proto.ColArr{
	Data: &tagsKeyLC,
}

Result: proto.Results{
	{Name: "tags.key", Data: &tagsKeyArr},
},
```

I get the following error

```
panic: decode block: decode block: target: tags.key: decode data: got version 30, expected 1

```
any example?

**Comment by ernado** [comment]
Now you can see TestArrLowCardinalityStr:
```go
str := &proto.ColStr{}
idx := &proto.ColLowCardinality{Index: str}
col := &proto.ColArr{Data: idx}
```

Use `col` to read of `Array(LowCardinality(String))` column.

Ergonomics will improve soon with go1.18 and generics.

**Comment by kant777** [comment]
@ernado so there are code changes? I need to download the latest code? looks like I am using the correct the syntax..Also another question is general. I see Result and onResult..do I need to put the same code for Result and OnResult if I were to read to all the blocks that are available as a result of my query?

**Reply by ernado** [reply]
Oh, can you please provide a full test case like we have in `query_test.go`?

You need to set OnResult when you expect more that single data block as query result.

**Reply by ernado** [reply]
Looks like there is a bug.
https://github.com/go-faster/ch/issues/30

Please usse https://github.com/ClickHouse/clickhouse-go for now, should be easier.

**Reply by ernado** [reply]
Should work in `v0.18.2`
